feat(Vehicle System): Vehicle system

PignataroLucas opened the pull request:
## Issue Ticket - 3

## Problem

- Ticket Header: *Vehicle System*
- Every marine can enter a vehicle
- Each vehicle have its maximumCapacity
- Only the pilot marine can drive the vehicle
- Specific vehicle bvehaviour
- Every marine can get out of the vehicle

## [Config] added or modified
```
Hay una escena para testear esto, SANDBOX

``` 

## TODO

- Hay que mejorar el sistema de movimiento del auto para hacerlo ver mas real.
- Actualmente si dos pilots estan subidos a un auto , no se puede seleccionar a un auto para manejarlo individualmente.
- Los marines en general no pueden salir del auto.
- No se que pasa si mandas un piloto a que se suba a un auto si ya hay un piloto conduciendo.

PignataroLucas committed:
feat(Vehicle System): Vehicle system first commit

PignataroLucas committed:
Update localization in folder interfaces

santihenry commented:
Por el momento el movimiento no importa, cumple la función que se necesita.

cuando tenes mas de un auto, con piloto se controlas todos juntos y no podes usarlos de manera individual, la causa de esto que el siguiente if:
![image](https://github.com/AgusRambal/SandWar/assets/22665134/949d8bbd-5896-42c2-8957-33b682ced75a)

solo se consulta si esta ON (tiene marines adentro), y eso es incorrecto se tiene que preguntar si esta seleccionado. para que de esa manera pueda seleccionar o deseleccionar a los autos como el user quiera.

Los autos no tienen limite de carga, entran todos los marine que quiera 
![image](https://github.com/AgusRambal/SandWar/assets/22665134/e579f4e7-769e-4114-96aa-aed11f69055c)

si mandas un piloto a que se suba a un auto que ya hay un piloto no pasa nada, se sube y funciona normal.

PignataroLucas commented:
> solo se consulta si esta ON (tiene marines adentro), y eso es incorrecto se tiene que preguntar si esta seleccionado. para que de esa manera pueda seleccionar o deseleccionar a los autos como el user quiera.
> Ahi va, barbaro, gracias por el feedback voy a estar trabajando con esto
> Los autos no tienen limite de carga, entran todos los marine que quiera ![image](https://user-images.githubusercontent.com/22665134/277802262-e579f4e7-769e-4114-96aa-aed11f69055c.png)
> Esto es un TODO, hagamos un issue aparte 
> si mandas un piloto a que se suba a un auto que ya hay un piloto no pasa nada, se sube y funciona normal.
Esto Tambien es un TODO

PD: Fijate de agregar los comentarios directamente a la linea de codigo O crear comentarios separados a cada observacion asi es mas prolijo

PignataroLucas commented:
Para dejar un comentario en la linea especifica o seccion especifica es apretando el + de aca 
![image](https://github.com/AgusRambal/SandWar/assets/91335382/7a5c938f-370a-44c6-96ae-8267b70b92b9)

santihenry commented on Assets/Scripts/VehicleSystem/InteractionHandler.cs:
```diff
@@ -0,0 +1,44 @@
+
+        public IUnit CurrentlySelectedUnit { get; private set; }
+
+        private void Update()
```
Lo que se hace en el Updete, hay que cambiarlo y que no sea si esta ON, tiene que funcionar como los marines que podes seleccionar o deseleccionar, y solo se mueve lo que esta seleccionado

PignataroLucas replied:
Vale

PignataroLucas committed:
Deselect and clean IH

Agregue la funcion de deseleccionar un auto sumado a que el bug de que los autos se mueven la mismo tiempo ya no ocurre

PignataroLucas committed:
Test1

PignataroLucas committed:
delet test script

PignataroLucas committed:
Check status

PignataroLucas committed:
delet test2

AgusRambal reviewed (approved):
De acuerdo con todo lo que dijo Santi, cuando este eso todo ok

PignataroLucas commented:
@santihenry  El commit que tenes que revisar es el Deselect and clean IH , tuve unos problemitas que tuve que solucionarlo con el bash, 
![image](https://github.com/AgusRambal/SandWar/assets/91335382/513976d1-8381-4424-9f64-2d803fe879f0)

PignataroLucas commented:
> De acuerdo con todo lo que dijo Santi, cuando este eso todo ok

Deja de robar

santihenry reviewed (commented):
Esta perfecto, solo le agregaría que te deje deseleccionar, porque una vez que tenes un auto ya no lo podes soltar, si alternar entre auto y auto, pero no dejar de usarlo

santihenry reviewed (approved):
Todo bien, mandale

PignataroLucas commented:
> Esta perfecto, solo le agregaría que te deje deseleccionar, porque una vez que tenes un auto ya no lo podes soltar, si alternar entre auto y auto, pero no dejar de usarlo

Es verdad, voy a crear una issue para eso
